File: app/src/main/java/com/waveneuro/ui/dashboard/history/DevicesHistoryAdapter.java
package com.waveneuro.ui.dashboard.history;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.waveneuro.R;
import com.waveneuro.data.model.response.device.SonalDevicesResponse;

import java.util.ArrayList;

public class DevicesHistoryAdapter extends RecyclerView.Adapter<ViewHolder> {

    Context context;
    private ArrayList<SonalDevicesResponse.Device> devices;

    public DevicesHistoryAdapter(Context context, ArrayList<SonalDevicesResponse.Device> devices) {
        this.context = context;
        this.devices = devices;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View v = (View) LayoutInflater.from(parent.getContext()).inflate(R.layout.devices_history_item, parent, false);

        return new ViewHolder(v);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        SonalDevicesResponse.Device device = devices.get(position);

        holder.deviceName.setText(device.getDeviceName());
        holder.deviceLastConnection.setText(device.getLastConnection());
    }

    @Override
    public int getItemCount() {
        return devices.size();
    }
}

File: app/src/main/java/com/waveneuro/ui/dashboard/history/adapter/DevicesHistoryAdapter.kt
package com.waveneuro.ui.dashboard.history.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.waveneuro.databinding.DevicesHistoryItemBinding
import com.waveneuro.domain.model.device.Device

class DevicesHistoryAdapter(
    private val context: Context,
    private val devices: List<Device>
) : RecyclerView.Adapter<DevicesHistoryAdapter.ViewHolder>() {

    inner class ViewHolder(val binding: DevicesHistoryItemBinding) : RecyclerView.ViewHolder(binding.root) {
        fun bind() {
            val device = devices[bindingAdapterPosition]

            with(binding) {
                tvDeviceName.text = device.deviceName
                tvDeviceLastConnection.text = device.lastConnection
            }
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder =
        ViewHolder(DevicesHistoryItemBinding.inflate(LayoutInflater.from(context), parent, false))

    override fun onBindViewHolder(viewHolder: ViewHolder, position: Int) {
        viewHolder.bind()
    }

    override fun getItemCount(): Int = devices.size

}
